soci/index_builder.go

```diff
@@ -0,0 +1,161 @@
+package soci
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+type IndexBuilder struct {
+	contentStore        content.Store
+	spanSize            int64
+	minLayerSize        int64
+	buildToolIdentifier string
+}
+
+func NewIndexBuilder(contentStore content.Store) *IndexBuilder {
+	defaultSpanSize := int64(1 << 22)
+	return &IndexBuilder{
+		contentStore:        contentStore,
+		spanSize:            defaultSpanSize,
+		minLayerSize:        0,
+		buildToolIdentifier: "AWS SOCI CLI v0.1",
```

Comment:
We probably should have the const string "AWS SOCI CLI v0.1"d defined else where and have the constructor pass in ```buildToolIdentifier```.

Reply:
yes will do in the next revision